src/services/tx/txSender.ts

```diff
@@ -192,17 +195,29 @@ export const dispatchTxSigning = async (
 /**
  * On-Chain sign a transaction
  */
-export const dispatchOnChainSigning = async (safeTx: SafeTransaction) => {
+export const dispatchOnChainSigning = async (safeTx: SafeTransaction, chain?: ChainInfo, txId?: string) => {
   const sdk = getAndValidateSafeSDK()
+  const provider = getWeb3()
+
+  if (!provider || !chain) return safeTx
```

Comment:
So the `chain` param is only passed for this check? Can we do it outside of the call instead?

Reply:
It was a leftover from a previous implementation. I removed it and also moved the `provider` check outside. Thanks!